build/builder.go

```diff
@@ -510,12 +535,24 @@ func NewBuilder(opts Options) (*Builder, error) {
 	return b, nil
 }
 
+// MarkFileAsChangedOrRemoved indicates that the file specified by the given path
+// has been changed or removed since the last indexing job for this repository.
+//
+// If this build is a delta build, these files will be tombstoned in the older shards for this repository.
+func (o *Options) MarkFileAsChangedOrRemoved(path string) {
```

Comment:
design: this should live on Builder, not Options. Same with the field. Think of it like the documents.